### PR TITLE
Add payload validation tests

### PR DESCRIPTION
**Pull request by jaymell**
## Motivation

Reach full test coverage of validation code.

## Solution

Add missing tests of payload validation logic.

### CODE REVIEW

File: server/svix-server/src/v1/endpoints/message.rs [hunk 1]
```diff
@@ -266,3 +266,81 @@ pub fn router() -> Router {
...
+    const INVALID_CHANNEL: &str = "$$invalid-channel";
+    const VALID_CHANNEL: &str = "valid-channel";
+    const INVALID_EVENT_TYPE: &str = "$$invalid-eventType";
+    const VALID_EVENT_TYPE: &str = "valid-eventType";
+    const INVALID_EVENT_ID: &str = "$$invalid-eventId";
+    const VALID_EVENT_ID: &str = "valid-eventId";
+    const INVALID_EVENT_CHANNELS: &[&str] = &["valid-event-channel", "&&invalid-event-channel"];
+    const VALID_EVENT_CHANNELS: &[&str] = &["valid-event-channel1", "valid-event-channel2"];
+
+    /*
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/message.rs:285]: 🤔

Comment (jaymell) [reply server/svix-server/src/v1/endpoints/message.rs:285]: Will clean that up :stuck_out_tongue:

**Review by svix-daniel — CHANGES_REQUESTED** [review]
All of this is repetitive -- I'd maybe do something like loops over test case elements and do the assertion once.

```rs
for version in [VERSION_VALID, VERSION_INVALID] {
    for url  in [URL_VALID, URL_INVALID] {
         for rate_limit in [None, Some(RATELIMIT_VALID),  Some(RATELIMIT_INVALID)]  {
               ...
         }
    }
}
```
or even 

```rs
for  endpoint_in in  [ep_1, ep_2, ep_3] {
     ....
}
```
for all the ones you expect to fail, then one  for what you expect to pass

File: server/svix-server/src/v1/endpoints/message.rs [hunk 2]
```diff
@@ -266,3 +266,66 @@ pub fn router() -> Router {
...
+        .unwrap();
+
+        let valid_3: MessageIn = serde_json::from_value(json!({
+            "eventType": EVENT_TYPE_VALID,
+            "payload": {},
+            "channels": EVENT_CHANNELS_INVALID
+        }))
+        .unwrap();
+
+        for m in [valid_1, valid_2, valid_3] {
```
Comment (svix-daniel) [inline server/svix-server/src/v1/endpoints/message.rs:307]: Should these not be `invalid_1` and  so  on?

**Review by svix-daniel — DISMISSED** [review]

File: server/svix-server/src/v1/endpoints/message.rs (hunk 2 above)
Comment (svix-james) [reply server/svix-server/src/v1/endpoints/message.rs]: Fixed.

**Review by svix-daniel — APPROVED** [review]

**Pull request merged**